src/test/java/DirectionsServletTest.java

```diff
@@ -85,22 +95,66 @@ public void clear() {
   }
 
   @Test
-  public void montrealToMontrealWithWaypoints() throws IOException {
-    
-    when(directionsClient.getDirections(Mockito.anyString(), Mockito.anyString(), Mockito.any(String[].class))).thenReturn(null);
+  public void windsorToMontrealWithWaypoints() throws Exception {
+    // Get optimized route of travel from Windsor, ON to Montreal, QC with waypoints Markham, ON,
+    // Waterloo, ON and Quebec City, QC in between.
+    // Should return four legs: Windsor, ON -> Waterloo, ON -> Markham, ON -> Quebec City, QC ->
+    // Montreal, QC.
+    when(directionsClient.getDirections(
+            Mockito.anyString(), Mockito.anyString(), Mockito.any(String[].class)))
+        .thenReturn(windsorToMontrealWithWaypoints);
     servlet.doGet(request, response);
     printWriter.flush();
-    Assert.assertTrue(stringWriter.toString().contains(NO_MESSAGES_JSON));
+    // Replace unicode and escaped characters
+    String actual =
+        stringWriter
+            .toString()
+            .replace("\\u003e", ">")
+            .replace("\\u003d", "=")
+            .replace("\\\"", "\"");
+    String expected = WINDSOR_TO_MONTREAL_WITH_WAYPOINTS_JSON;
```

Comment:
Just use the constant name directly - that's why you declared those fantastic constants to begin with!

Reply:
Yeah you're absolutely right, thanks!

Reply:
`RESOLVED`